bundles/admin/admin-hierarchical-layerlist/instance.js

```diff
@@ -0,0 +1,364 @@
+            });
+        },
+        /**
+         * Add layertree options
+         * @method  _addOptions
+         * @private
+         */
+        _addOptions: function() {
+            var me = this;
+
+            // Add Drag & drop plugin
+            me.service.addLayerlistOption('plugins', ['checkbox', 'changed', 'wholerow', 'types', 'search', 'state', 'conditionalselect', 'dnd'], false);
+            me.service.addLayerlistOption('dnd', {
+                use_html5: true,
+                inside_pos: 'last'
+            });
+        },
+
+        /**
+         * Add layer tools
+         * @method  _addLayerTools
+         * @private
+         */
+        _addLayerTools: function() {
+            var me = this;
+            // Add edit tool for layers
+            me.service.addLayerTool('edit-layer', function(tool, groupId, layerId) {
+
+                me.layer.showLayerAddPopup(tool, layerId, groupId);
+            }, {
+                cls: 'edit-layer',
+                tooltip: me.locale('tooltips.editLayer')
+            });
+        },
+
+        /**
+         * Assigns event handlers to the admin hierarchical layer list.
+         */
+        _addEventHandlers: function() {
+            var me = this;
+            jQuery(document).on("dnd_stop.vakata", function(event, data) {
+                //If the drag target group is not open, we have to open it.
+                //Otherwise we can't get the necessary information of the drag operation.
+                var targetGroup = data.data.origin.get_node(jQuery(data.event.target).prop("id").split("_")[0]);
+                if (!data.data.origin.is_open(targetGroup)) {
+                    data.data.origin.open_node(targetGroup);
+                    me._saveOrder(event, data);
+                } else {
+                    me._saveOrder(event, data);
+                }
+            });
+        },
+        /**
+         * Find the nodes actual type which can be either layer or group (includes subgroups and subgroup-subgroups)
+         * @param  {String} type The node type in the tree
+         * @return {String}      The node actual type - layer or group.
+         */
+        _findJSTreeNodeActualType: function(type) {
+            var actualType;
+            if (type.indexOf('group') >= 0) {
+                actualType = 'group';
+            } else {
+                actualType = 'layer';
+            }
+            return actualType;
+        },
+
+        /**
+         * Save group or list ordering and grouping.
+         * @method  _saveOrder
+         * @param   {Object}   data  data for saving
+         * @param   {Oskari.userinterface.component.Popup}   popup group adding/editing popup
+         * @param   {String}   type  jstree type
+         * @private
+         */
+        _saveOrder: function(event, data) {
+            var me = this;
+            var draggedNode = data.data.origin.get_node(data.data.nodes[0]);
```

Comment:
Does data need to be passed around or could you just send draggedNode as param?

Reply:
You should restrict the usage of "data" and event param so they are not needed here. Dig out the information you need in the calling function and pass it here as param.